src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -509,7 +555,12 @@ impl ExprRewriter for LogicalAggBuilder {
     /// Note that the rewriter does not traverse into inputs of agg calls.
     fn rewrite_agg_call(&mut self, agg_call: AggCall) -> ExprImpl {
         let return_type = agg_call.return_type();
-        let (agg_kind, inputs, distinct, order_by, filter) = agg_call.decompose();
+        let (agg_kind, inputs, distinct, mut order_by, filter) = agg_call.decompose();
+        if agg_kind != AggKind::StringAgg {
```

Comment:
To be conservative, we should treat newly added AggKind in the future as not rewritable.